streams/src/test/java/org/apache/kafka/streams/integration/EosV2UpgradeIntegrationTest.java

```diff
@@ -947,8 +948,9 @@ public void process(final Record<Long, Long> record) {
             } else {
                 int exceptionCount = exceptionCounts.get(appDir);
                 // should only have our injected exception or commit exception, and 2 exceptions for each stream
-                if (++exceptionCount > 2 || !(e instanceof RuntimeException) ||
-                    !(e.getMessage().contains("test exception"))) {
+                if (++exceptionCount > 2 ||
+                    !(e instanceof StreamsException) ||
+                    !(e.getCause().getMessage().endsWith(" test exception."))) {
```

Comment:
As above. However, we cannot use `equals` as the use different error messages for different situation when we throw.